packages/central-server/app/sync/CentralSyncManager.js

```diff
@@ -647,16 +647,18 @@ export class CentralSyncManager {
           { direction: SYNC_SESSION_DIRECTION.INCOMING },
         );
 
+        // Tick tock once more to ensure that no records that are subsequently modified will share the same sync tick as the incoming changes
+        // notably so that if records are modified by adjustDataPostSyncPush(), they will be picked up for pulling in the same session
+        // (specifically won't be removed by removeEchoedChanges())
+        await this.tickTockGlobalClock();
+
         return tock;
       });
 
       await models.SyncDeviceTick.create({
         deviceId,
         persistedAtSyncTick,
       });
-      // tick tock global clock so that if records are modified by adjustDataPostSyncPush(),
-      // they will be picked up for pulling in the same session (specifically won't be removed by removeEchoedChanges())
-      await this.tickTockGlobalClock();
```

Comment:
Previously it was this tickTock that might've been blocked (perhaps by another sync) which meant there was a reasonable time window in which other changes on the central-server might occur and share the sync tick with the synced records.

Decided to cut it as now we're tickTocking in the transaction anyway so there's no need to do it here.